src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceRequest.java

```diff
@@ -0,0 +1,147 @@
+ */
+@Getter
+@Setter
+@Log4j2
+public class PutDatasourceRequest extends AcknowledgedRequest<PutDatasourceRequest> {
+    private static final ParseField ENDPOINT_FIELD = new ParseField("endpoint");
+    private static final ParseField UPDATE_INTERVAL_IN_DAYS_FIELD = new ParseField("update_interval_in_days");
+    /**
+     * @param datasourceName the datasource name
+     * @return the datasource name
+     */
+    private String datasourceName;
+    /**
+     * @param endpoint url to a manifest file for a datasource
+     * @return url to a manifest file for a datasource
+     */
+    private String endpoint;
+    /**
+     * @param updateIntervalInDays update interval of a datasource
+     * @return update interval of a datasource
+     */
+    private TimeValue updateIntervalInDays;
+
+    /**
+     * Parser of a datasource
+     */
+    public static final ObjectParser<PutDatasourceRequest, Void> PARSER;
+    static {
+        PARSER = new ObjectParser<>("put_datasource");
+        PARSER.declareString((request, val) -> request.setEndpoint(val), ENDPOINT_FIELD);
+        PARSER.declareLong((request, val) -> request.setUpdateIntervalInDays(TimeValue.timeValueDays(val)), UPDATE_INTERVAL_IN_DAYS_FIELD);
+    }
+
+    /**
+     * Default constructor
+     * @param datasourceName name of a datasource
+     */
+    public PutDatasourceRequest(final String datasourceName) {
+        this.datasourceName = datasourceName;
+    }
+
+    /**
+     * Constructor with stream input
+     * @param in the stream input
+     * @throws IOException IOException
+     */
+    public PutDatasourceRequest(final StreamInput in) throws IOException {
+        super(in);
+        this.datasourceName = in.readString();
+        this.endpoint = in.readString();
+        this.updateIntervalInDays = in.readTimeValue();
+    }
+
+    @Override
+    public void writeTo(final StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(datasourceName);
+        out.writeString(endpoint);
+        out.writeTimeValue(updateIntervalInDays);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException errors = new ActionRequestValidationException();
+        validateEndpoint(errors);
+        validateUpdateInterval(errors);
+        return errors.validationErrors().isEmpty() ? null : errors;
+    }
+
+    private void validateEndpoint(final ActionRequestValidationException errors) {
+        try {
+            URL url = new URL(endpoint);
+            url.toURI(); // Validate URL complies with RFC-2396
+            validateManifestFile(url, errors);
+        } catch (MalformedURLException | URISyntaxException e) {
+            log.info("Invalid URL format is provided", e);
+            errors.addValidationError("Invalid URL format is provided");
+        }
+    }
+
+    private void validateManifestFile(final URL url, final ActionRequestValidationException errors) {
```

Comment:
Explain in java doc or in code comments why every validation is required. It will make sure that we know before removing any validation why it is required.